marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java

```diff
@@ -560,7 +560,7 @@ private void addToBatcher(File file, Format fileFormat) throws  IOException{
                 Stream<JacksonHandle> contentStream = splitter.split(docStream);
                 String uri = generateAndEncodeURI(file.getParent());
                 Stream<DocumentWriteOperation> documentStream =  DocumentWriteOperation.from(
-                    contentStream, DocumentWriteOperation.uriMaker(outputURIReplace(uri)+"/%s.json"));
+                    contentStream, DocumentWriteOperation.uriMaker(outputURIReplace(uri).replace("%", "%%")+"/%s.json"));
```

Comment:
Are you sure it will fix 2221? Url encode for '%' -> %25
2221 exception said:
java.util.IllegalFormatConversionException: c != java.util.UUID\r\n\tat 
looks like "return String.format(uriFormat, UUID.randomUUID());" in DocumentWriteOperation class caused the exception...

Reply:
I think it should.. uriFormat is userString+ "/%s.json" , so am escaping "%" in userString. https://stackoverflow.com/questions/5011932/how-to-escape-in-string-format

Reply:
but I guess that the url doesn't include '%', it is a path. I feel the java api may need to change -> String.format(uriFormat, UUID.randomUUID().toString())

Reply:
I am encoding the abs path to uri and in Windows, "\" gets encoded to "%5C" and hence we see the issue in Windows.

Reply:
dir you have a chance to test it?

Reply:
yes, it was tested on Timur's windows machine